spec/support/macros.rb

```diff
@@ -0,0 +1,23 @@
+def current_user
+  User.find(session[:user_id])
+end
+
+def set_user(user=nil)
+  session[:user_id] = (user || Fabricate(:user)).id
+end
+
+def signout_user
+  session[:user_id] = nil
+end
+
+def user_logged_in?
+  session[:user_id] != nil
+end
+
+def sign_in(a_user=nil)

```

Comment:
:/ Brain overloaded of concepts. Starting to delete things from it. Deleted 5 seasons of lost and it seems i continued with memoization :). I´ll try to remember next time. Sorry for this.